packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1430,11 +1463,35 @@ export function attach(
     nextFiber: Fiber,
     prevFiber: Fiber,
     parentFiber: Fiber | null,
+    traceNearestHostComponentUpdate: boolean,
   ): boolean {
     if (__DEBUG__) {
       debug('updateFiberRecursively()', nextFiber, parentFiber);
     }
 
+    if (traceUpdatesEnabled) {
+      const elementType = getElementTypeForFiber(nextFiber);
+      if (traceNearestHostComponentUpdate) {
+        // If an ancestor updated, we should mark the nearest host nodes for highlighting.
+        if (elementType === ElementTypeHostComponent) {
+          traceUpdatesForNodes.add(nextFiber.stateNode);
+          traceNearestHostComponentUpdate = false;
+        }
+      } else {
+        if (
+          elementType === ElementTypeFunction ||
+          elementType === ElementTypeClass ||
+          elementType === ElementTypeContext
```

Comment:
What if `traceNearestHostComponentUpdate` is false because the parent didn't rerender, but the props to the host component has changed. Seems like you'd want to include `ElementTypeHostComponent` here too, right?

Reply:
How would the props to a host component change without a parent component re-rendering?

Reply:
Good question. I guess it's special cases like: `<div hidden="foo"><div prop={...} /></div>`  The prop doesn't change in the same commit as the parent rerendering.

Also for nested host  components: `Foo = ({change}) => <div><div prop={change} /></div>`

The outer parent rerenders but the "traceNearestHostComponentUpdate" stops at the outer host component but you want to mark the inner one as having changed.

Reply:
WRT the nested example above, I think it still makes sense to highlight the outermost node, because that's the one associated with the `Foo` component (which is the thing that's actually re-rendering).

I think this makes sense because we highlight the outer `<div>` in the case where `Foo` re-renders but doesn't cause any host effects at all.

Reply:
Chatted out of band and agreed that the current behavior is okay.

In the first case, it's likely that a nested component component would also re-render and so we would highlight its nodes.

In the second case, highlighting the outer `<div>` only is probably sufficient because this is just meant to be a coarse signal that a composite component re-rendered- not a fine grained tool like the Profiler.